sdk/eventhub/azure-eventhub/azure/eventhub/_buffered_producer/_buffered_producer.py

```diff
@@ -140,10 +141,15 @@ def flush(self, timeout_time=None, raise_error=True):
             _LOGGER.info("Partition: %r started flushing.", self.partition_id)
             if self._cur_batch:  # if there is batch, enqueue it to the buffer first
                 self._buffered_queue.put(self._cur_batch)
-            while self._cur_buffered_len:
+            while self._buffered_queue.qsize() > 0:
                 remaining_time = timeout_time - time.time() if timeout_time else None
                 if (remaining_time and remaining_time > 0) or remaining_time is None:
-                    batch = self._buffered_queue.get()
+                    try:
+                        batch = self._buffered_queue.get(block=False)
+                    except queue.Empty:
+                        #reset buffered count
+                        self._cur_buffered_len = 0
```

Comment:
We're only going to enter this `except` clause in the odd situation that the `while qsize() > 0` loop continued to iterate yet it still raised an Empty. So we should probably reset the `cur_buffered_len` outside of the loop?

Reply:
I can do that, the only reason I had it over there was because if the while loop exited normally then `cur_buffered_len` would be 0 already.